packages/vit-s/src/generate-alt-text.test.js

```diff
@@ -0,0 +1,72 @@
+import { arrayToString, getViewTypesMap, getAltText } from './generate-alt-text.js';
+
+describe('Tests for generating an alt text from the vitessce config', () => {
+  describe('Tests for arrayToString', () => {
+    it('should return empty string for empty array', () => {
+      const arr = [];
+      const str = arrayToString(arr);
+      expect(str).toEqual('');
+    });
+
+    it('should return the element if the araay only has one element', () => {
+      const arr = ['one'];
+      const str = arrayToString(arr);
+      expect(str).toEqual('one');
+    });
+
+    it('should return the elements concatenated with "and" when the array has two elements', () => {
+      const arr = ['one', 'two'];
+      const str = arrayToString(arr);
+      expect(str).toEqual('one and two');
+    });
+
+    it('should return the elements concatenated with ", " and the last two elements with "and" when the array has more than two elements', () => {
+      const arr = ['one', 'two', 'three', 'four'];
+      const str = arrayToString(arr);
+      expect(str).toEqual('one, two, three and four');
+    });
+  });
+
+  describe('Tests for getViewTypesMap', () => {
+    it('should have a natural language mapping for every defined viewtype', () => {
+      const mapping = getViewTypesMap();
+      const mappingLength = mapping.keys().length;
+      expect(mappingLength).toEqual(ViewType.length);
+    });
+
+    it('should not have undefined in the mapping', () => {
+      const mapping = getViewTypesMap();
+      expect(undefined in mapping.keys() || undefined in mapping.values()).toEqual(false);
+    });
+  });
+
+  describe('Tests for getAltTest', () => {
+    it('should return a string', () => {
+      const config = {
+        coordinationSpace: {},
+        datasets: {},
+        description: 'Spatial organization of the somatosensory cortex revealed by osmFISH',
+        initStrategy: 'auto',
+        layout: [
+          { component: 'description', props: { description: 'Codeluppi et al., Nature Methods 2018: Spatial organization of the somatosensory cortex revealed by osmFISH' }, x: 0, y: 0, w: 2 },
+          { component: 'layerController', x: 0, y: 1, w: 2, h: 4 },
+          { component: 'status', x: 0, y: 5, w: 2, h: 1 },
+          { component: 'spatial', props: { channelNamesVisible: true }, x: 2, y: 0, w: 4 },
+          { component: 'featureList', x: 9, y: 0, w: 3, h: 2 },
+          { component: 'obsSets', x: 9, y: 3, w: 3, h: 2 },
+          { component: 'heatmap', props: { transpose: true }, x: 2, y: 4, w: 5 },
+          { component: 'obsSetFeatureValueDistribution', x: 7, y: 4, w: 5, h: 2 },
+          { component: 'scatterplot', x: 6, y: 0, w: 3, h: 2 },
+          { component: 'scatterplot', x: 6, y: 2, w: 3, h: 2 },
+        ],
+        name: 'Codeluppi et al., Nature Methods 2018',
+        uid: 'A',
+        version: '1.0.16',
+      };
+      const altText = getAltText(config);
+      expect(typeof altText).toEqual('string');
```

Comment:
```suggestion
      expect(typeof altText).toEqual('string');
      // eslint-disable-next-line max-len
      expect(altText).toEqual('Vitessce grid with 10 views, including description, layer controller, status view for debugging, spatial view, interactive list of features, list of potential observation sets, cell by gene heatmap, violin plot with values, scatterplot with embeddings and scatterplot with embeddings.');
```

We can also check the content of the string

Reply:
I agree with checking the context, but perhaps not with the direct result, because then adding new features would break the test. Instead, I added two tests that check if the number of views is mentioned and that a few of the view types are included!

Reply:
That makes sense, thanks!